pipelines/batch/src/main/java/com/google/fhir/analytics/ParquetMerger.java

```diff
@@ -64,31 +64,50 @@ public class ParquetMerger {
   private static String SYSTEM_KEY = "system";
   private static String CODE_KEY = "code";
 
-  private static PCollection<KV<String, GenericRecord>> readAndMapToId(
-      Pipeline pipeline, DwhFiles dwh, String resourceType) {
+  private static PCollection<KV<String, Iterable<GenericRecord>>> readAndGroupById(
+      Pipeline pipeline, List<DwhFiles> dwhFilesList, String resourceType) {
+
+    PCollection<ReadableFile> inputFiles =
+        pipeline
+            .apply(Create.of(getParquetFilePaths(resourceType, dwhFilesList)))
+            .apply(FileIO.matchAll())
+            .apply(FileIO.readMatches());
```

Comment:
I have verified the number of the parallel worker threads that will be used after this change and it remains the same as before. I am assuming performance tests should not be part of the CI as it will delay the build. I am open for suggestions on this.

Reply:
We can have a non-blocking performance test, i.e., it won't block a merge but we run it on each commit such that if a performance regression is introduced we can pinpoint the culprit quickly; but that's for #607. For this change if you can run the pipeline on a larger input, e.g., the 8K Patient set (with 1.6M Observations) and compare before/after run-time, that would be great.

Reply:
Benchmarking is in progress, will update the results once done.

Reply:
I have attached the benchmark results in the PR description. The performance seems to have improved after fixing the regression.